options/options.go

```diff
@@ -0,0 +1,37 @@
+package options
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero"
+)
+
+type WazeroConfigOption func(plugin *WazeroConfig)
+
+type WazeroNewRuntime func(context.Context) (wazero.Runtime, error)
+
+type WazeroConfig struct {
+	NewRuntime   func(context.Context) (wazero.Runtime, error)
```

Comment:
As these are exported, I wonder if we shouldn't just pass the struct instead of the options pattern. wdyt style-wise @achille-roussel